Gradient computation is incorrect if a node in expression graph is formed by gorgonia.Neg operator

The gradient gets set to zero incorrectly if the last node in the expression graph is formed by ``` gorgonia.Neg``` operator. The gradients are correct if instead multiplication by 1 operator is used.

Here is the test which illustrates this problem.
```go
func TestNegOp(t *testing.T) {
	g := gorgonia.NewGraph()

	x := gorgonia.NewScalar(g, gorgonia.Float64, gorgonia.WithName("x"))
	y := gorgonia.NewScalar(g, gorgonia.Float64, gorgonia.WithName("y"))

	z, err := gorgonia.Add(x, y)

	if err != nil {
		t.Fatal("error adding:", err)
	}

	z, err = gorgonia.Neg(z) //last node is Neg operator. So gradients are zero
       
       // z, err = gorgonia.Mul(gorgonia.NewConstant(-1.0), z) //This results in non zero gradients
       

	if err != nil {
		t.Fatal("error in Multiply with -1:", err)
	}

	gorgonia.Let(x, 2.5)
	gorgonia.Let(y, 2.0)

	m := gorgonia.NewLispMachine(g)

	defer m.Close()

	err = m.RunAll()

	if err != nil {
		t.Fatal("error in running the lisp machine:", err)
	}

	t.Log("value of z:", z.Value())

	xgrad, err := x.Grad()

	if err != nil {
		t.Fatal("error in getting the xgrad:", err)
	}

	ygrad, err := y.Grad()

	if err != nil {
		t.Fatal("error in getting the ygrad:", err)
	}

	actualxgrad := xgrad.Data().(float64)

	actualygrad := ygrad.Data().(float64)

	if actualxgrad == 0.0 {
		t.Log("xgrad=", actualxgrad, "ygrad=", actualygrad)
		t.Fatal("zero xgrad")
	}

	if actualygrad == 0.0 {
		t.Fatal("zero ygrad")
	}

	t.Log("xgrad=", actualxgrad, "ygrad=", actualygrad)

}

```

Test Output:
```bash
=== RUN   TestNegOp
--- FAIL: TestNegOp (0.00s)
    mledemo_test.go:100: value of z: -4.5
    mledemo_test.go:119: xgrad= 0 ygrad= 0
    mledemo_test.go:120: zero xgrad
FAIL
exit status 1
FAIL    mledemo 0.011s
```

--- a/known_issues_test.go
+++ b/known_issues_test.go
@@ -235,5 +235,68 @@ func TestIssue233_F64(t *testing.T) {
 	}
 
 	assert.Equal(t, correct, y.Value().Data())
+}
+
+func TestIssue363(t *testing.T) {
+	g := NewGraph()
+
+	x := NewScalar(g, Float64, WithName("x"))
+	y := NewScalar(g, Float64, WithName("y"))
+
+	z, err := Add(x, y)
+
+	if err != nil {
+		t.Fatal("error adding:", err)
+	}
+
+	z, err = Neg(z) //last node is Neg operator. So gradients are zero
+
+	// z, err = gorgonia.Mul(gorgonia.NewConstant(-1.0), z) //This results in non zero gradients
+
+	if err != nil {
+		t.Fatal("error in Multiply with -1:", err)
+	}
+
+	Let(x, 2.5)
+	Let(y, 2.0)
+
+	m := NewLispMachine(g)
+
+	defer m.Close()
+
+	err = m.RunAll()
+
+	if err != nil {
+		t.Fatal("error in running the lisp machine:", err)
+	}
+
+	t.Log("value of z:", z.Value())
+
+	xgrad, err := x.Grad()
+
+	if err != nil {
+		t.Fatal("error in getting the xgrad:", err)
+	}
+
+	ygrad, err := y.Grad()
+
+	if err != nil {
+		t.Fatal("error in getting the ygrad:", err)
+	}
+
+	actualxgrad := xgrad.Data().(float64)
+
+	actualygrad := ygrad.Data().(float64)
+
+	if actualxgrad == 0.0 {
+		t.Log("xgrad=", actualxgrad, "ygrad=", actualygrad)
+		t.Fatal("zero xgrad")
+	}
+
+	if actualygrad == 0.0 {
+		t.Fatal("zero ygrad")
+	}
+
+	t.Log("xgrad=", actualxgrad, "ygrad=", actualygrad)
 
 }

--- a/operatorPointwise_unary.go
+++ b/operatorPointwise_unary.go
@@ -311,10 +311,21 @@ func negDiff(x, y *Node) (err error) {
 	xdv, ydv := getDV(x, y)
 
 	sub := newElemBinOp(subOpType, x, y)
-	_, err = sub.UnsafeDo(xdv.d, ydv.d)
+	var d Value
+	d, err = sub.UnsafeDo(xdv.d, ydv.d)
+
+	// first we check if what essentially is a noIncrError is called
 	if err = checkErrSetDeriv(err, xdv); err != nil {
 		return errors.Wrapf(err, autodiffFail, x)
 	}
+
+	// then we set derivs, if d is a scalar
+	if _, ok := xdv.Value.(Scalar); ok {
+		if err = xdv.SetDeriv(d); err != nil {
+			return errors.Wrapf(err, autodiffFail, x)
+		}
+	}
+
 	return
 }
 

